pkg/postgres/pgutils/retry.go

```diff
@@ -0,0 +1,37 @@
+package pgutils
+
+import (
+	"time"
+)
+
+const (
+	interval = 5 * time.Second
+	timeout  = 5 * time.Minute
+)
+
+// RetryQuery is used to specify how long to retry to successfully run a query
+// that fails with Transient errors
+func RetryQuery(fn func() error) error {
+	// Run query immediately
+	if err := fn(); !isTransientError(err) {
+		return err
+	}
+
+	expiration := time.NewTimer(timeout)
+	defer expiration.Stop()
+
+	intervalTicker := time.NewTicker(interval)
+	defer intervalTicker.Stop()
+
+	var err error
+	for {
+		select {
+		case <-expiration.C:
+			log.Fatalf("unsuccessful in reconnecting to the database: %v. Exiting...", err)
+		case <-intervalTicker.C:
+			if err = fn(); !isTransientError(err) {
```

Comment:
Same here. Also, remove the declaration in l.26 in favor of a more tightly-scoped `err`:
```suggestion
			if err := fn(); err == nil || !isTransientError(err) {
```

Reply:
The declaration in line 26 allows the final error to be logged in case of expiration

Reply:
👍